### ISSUE TITLE
Find random post to comment/like/reshare

### ISSUE THREAD

**Issue by schiessle** [friendica/friendica#7633]
It is the nature of the fediverse that a lot of stuff happens outside of your own instance. It happens to me quite often that I find a post somewhere which I would like to comment/reshare/like but it is not in my timeline because it is no one I follow.

Mastodon has this really nice feature that I can enter a direct link to a post in the search, the post will then be shown as a search result and I can easily comment, reshare it or like it.

Does something like this also exists for Friendica or is it planed?

**Comment by annando** [comment]
Yes, just copy the url into the search bar. The post will then be fetched. This works for both ActivityPub and Diaspora.

There is a problem though with Pleroma, since their link to the post doesn't represent or redirect to the AP URL.

BTW: You need the current RC for this.

**Comment by schiessle** [comment]
> BTW: You need the current RC for this.

Ah, OK. I'm on the latest stable... That's why it probably didn't worked. But create that it is already there! 🙂

**Comment by annando** [comment]
"create"? 😁

**Comment by MrPetovan** [comment]
Probably "great".

**Comment by SpcCw** [comment]
Are there some conditions for this feature to work? Today I found this and was very excited because it would allow me to import my existing postings from Diaspora and categorize them (even though categories wouldn't work directly, they would via reshares).

However some posts get fetched, some don't - even from the same source. Generally it seems that fresh posts get fetched and older ones are less likely to do so. I couldn't fetch anything older than my Friendica server itself at all.

Some examples:

https://diasp.org/posts/0356fc500faa01389f5800163e3b85e3 - ok
https://diasp.org/posts/dd4cd3900e060138bd37271e4cde6c2e - ok
https://diasp.org/posts/782eb7b0cb6d0137a01f2a0000053625 - not ok (but imports on mordor.social)
https://diasp.org/posts/e18176ef3a0e659a5b01fbb4d3e6f9890aa5f844 - not ok

https://deko.cloud/posts/0b0a58e00d530138cb541600003291ce - ok
https://deko.cloud/posts/db1db0400d540138cb541600003291ce - not ok
https://deko.cloud/posts/006fae200cb801381bfc1600003291ce - ok
https://deko.cloud/posts/cc7fbfb00c6601381bfa1600003291ce - ok
https://deko.cloud/posts/14c90b40d1a00137ba4816000018adb5 - not ok
https://deko.cloud/posts/bb6a6850d4bf0137ba4616000018adb5 - not ok
https://deko.cloud/posts/eedc2350d4ea0137ba4816000018adb5 - not ok

Not so good with Friendica as source too:
https://mordor.social/display/027b03f7-135e-0faa-af45-637226226705 - ok
https://mordor.social/display/027b03f7-175c-a368-6a0b-0be481544494 - not ok
https://mordor.social/display/027b03f7-555c-ab89-9cdc-47c375296523 - not ok

**Comment by annando** [comment]
We do have some date limit. Several of your posts are really old. So possibly you hit that limit with these.

**Comment by MrPetovan** [comment]
How does the date limit work?

**Comment by annando** [comment]
Partly reliable.

**Comment by SpcCw** [comment]
Guys, you are **_amazing_** 👍 

I set "Lifespan of remote items" to 0 in "Admin - Performance" and clear "Automatically expire posts after this many days" in user settings and now everything gets fetched.

Even stars and folders work if the author was added to contacts _before_ fetching their posts. This is great as it allows migrating blogs!

**Comment by MrPetovan** [comment]
Well this is counter-intuitive. Shouldn’t it be a lifespan since an item was received?

**Comment by annando** [comment]
This avoids fetching the same old feed item again and again and again.

**Comment by MrPetovan** [comment]
Can it be limited to feeds?

**Comment by annando** [comment]
This would also include Friendica contacts, since we do pull them as well. And when there was some "dead" Friendica account, it would fetch the same old content over and again as well.

**Comment by MrPetovan** [comment]
So this means our way of deleting an item isn't ideal. We want to keep a trace of what was deleted, like the URI, so that we don't fetch the same item again. We can remove the item content but keep the item row for that purpose.

**Comment by annando** [comment]
This would be bad, since this would lead to a slower and slower system over the years.

**Comment by MrPetovan** [comment]
I don't believe it would be significant. Even with a fresh database Friendica is pretty slow, so this wouldn't rank pretty high in the performance issues chart.

**Comment by annando** [comment]
It would be significant over time and one should never mix productive tables with archive tables. The item table is really wide as well, it has got many fields, also the same item is multiplied with the numbers of receivers, which additionally adds space. However, there are other possible solutions. Like the "userd" table there could be a table with the URI and GUID of deleted items.

**Comment by MrPetovan** [comment]
I wouldn’t mind an `itemd` table to ensure posts aren’t arbitrarily rejected.
